src/Tools/dotnet-monitor/CollectionRules/CollectionRulePipeline.cs

```diff
@@ -20,25 +20,29 @@ internal class CollectionRulePipeline : Pipeline
         // The executor of the action list for the collection rule.
         private readonly ActionListExecutor _actionListExecutor;
 
-        private readonly CollectionRuleContext _context;
+        public CollectionRuleContext Context { get; }
 
         // Task completion source for signalling when the pipeline has finished starting.
         private readonly Action _startCallback;
-        
+
         // Operations for getting trigger information.
         private readonly ICollectionRuleTriggerOperations _triggerOperations;
 
         // Flag used to guard against multiple invocations of _startCallback.
         private bool _invokedStartCallback = false;
 
+        public DateTime PipelineStartTime { get; private set; }
```

Comment:
Move this to CollectionRulePipelineState since it's not used by anything in CollectionRulePipeline.